enhancements/single-node-production-edge-cluster-profile.md

```diff
@@ -0,0 +1,469 @@
+
+ spec:
+   managementState: "Managed"
+   unsupportedConfigOverrides:
+     useUnsupportedUnsafeNonHANonProductionUnstableOAuthServer: true
+EOF
+)"
+```
+
+#### cluster-ingress-operator
+
+By default, the `cluster-ingress-operator` deploys the router with 2
+replicas. On a single node one will fail to start and the ingress will
+show as degraded.
+
+```shell
+# patch ingress operator to run a single router pod
+oc patch -n openshift-ingress-operator ingresscontroller/default --type=merge --patch '{"spec":{"replicas": 1}}'
+```
+
+#### machine-config-operator
+
+The machine-config-operator includes a check in code that the number
+of control plane nodes is 3. Removing this or changing the minimum is
+required to allow the operator to complete its work, for example to
+enable the realtime kernel on a single node with the appropriate
+performance profile settings.
+
+
+#### Other deployments
+
+All of these operators need to have their replica count set to 1.
+
+**Note**: The cluster-version-operator will restore these deployments
+back to 2 replicas.
+
+* openshift-authentication/oauth-openshift
+* openshift-console/console
+* openshift-console/downloads
+* openshift-monitoring/prometheus-adapter
+* openshift-monitoring/thanos-querier
+* openshift-operator-lifecycle-manager/packageserver
+
+
+
+### Risks and Mitigations
+
+*What are the risks of this proposal and how do we mitigate. Think broadly. For
+example, consider both security and how this will impact the larger OKD
+ecosystem.*
+
+*How will security be reviewed and by whom? How will UX be reviewed and by whom?*
+
+## Design Details
+
+### Open Questions
+
+1. Telco workloads frequently require a realtime kernel. How will a
+   user specify whether to use the realtime or regular kernel? Should
+   we assume they always want the realtime version?
+2. Similarly, telco workloads typically require special network setups
+   for a host to boot, including bonded interfaces, access to multiple
+   VLANs, and static IPs. How do we anticipate configuring those?
+3. The machine-config-operator works by (almost always) rebooting a
+   host.  Is that going to be OK in these single-node deployments?
+   MCO is used by the performance-addon-operator and the
+   network-tuning-operator to apply the computed host OS and kernel
+   tuning values.  It is also used to allocate hugepages.  Do we want
+   it to run in a different mode where reboots are not performed?
+
+### Test Plan
+
+In order to claim full support for this configuration, we must have
+CI coverage informing the release. An end-to-end job using the profile
+and running an appropriate subset of the standard OpenShift tests
+will be created and configured to block accepting release images
+unless it passes.
+
+That end-to-end job should also be run against pull requests for
+the operators and other components that are most affected by the new
+profile, such as the etcd and auth operators.
+
+### Graduation Criteria
```

Comment:
since this requires a bunch of non-trivial work and coordination would like to see what the rollout plans for this look like..